### PR TITLE
Add task to update topology in PD

### PR DESCRIPTION
**Pull request by mapleFU**
Signed-off-by: mapleFU <1506118561@qq.com>

### CODE REVIEW

File: pkg/task/update_topology.go [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+			DeployPath: instance.DeployDir(),
+		}
+		data, err := json.Marshal(topology)
+		if err != nil {
+			return nil, err
+		}
+		op := clientv3.OpPut("/topology/"+instance.ComponentName(), string(data))
+		return &op, nil
+	default:
+		panic("updateTopologyOp receive wrong arguments, logic error!")
```
Comment (AstroProfundis) [inline pkg/task/update_topology.go:89]: We should return error here instead of panic

File: pkg/task/update_topology.go [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+	_, err = txn.Then(ops...).Commit()
+	return err
+}
+
+// componentTopology represent the topology info for alertmanager, prometheus and grafana.
+type componentTopology struct {
+	IP         string `json:"ip"`
+	Port       int    `json:"port"`
+	DeployPath string `json:"deploy_path"`
```
Comment (AstroProfundis) [inline pkg/task/update_topology.go:71]: Is `DeployPath` necessary here? We generally use IP:Port to identify an instance.

**Comment by july2993** [comment]
pls fix the test
```
Checking service state of grafana
	172.19.0.101	   Active: active (running) since Thu 2020-04-30 12:00:32 UTC; 21s ago
Checking service state of alertmanager
	172.19.0.101	   Active: active (running) since Thu 2020-04-30 12:00:44 UTC; 10s ago
+ [ Serial ] - UpdateTopology: cluster=test_cmd
{"level":"warn","ts":"2020-04-30T12:00:54.722Z","caller":"clientv3/retry_interceptor.go:61","msg":"retrying of unary invoker failed","target":"endpoint://client-83becfcf-3c32-47d3-bc8f-6f1b4e250e22/172.19.0.103:2379","attempt":0,"error":"rpc error: code = InvalidArgument desc = etcdserver: duplicate key given in txn request"}

Error: etcdserver: duplicate key given in txn request

Verbose debug logs has been written to /tiup-cluster/tests/logs/tiup-cluster-debug-2020-04-30-12-00-54.log.
```

File: pkg/task/update_topology.go (hunk 2 above)
Comment (mapleFU) [reply pkg/task/update_topology.go:68]: Dashboard may want's to know every component's deploy path.

**Comment by mapleFU** [comment]
@july2993 used in tidb-dashboard to show cluster topology

**Comment by lonng** [comment]
Closed via #377

**Pull request closed without merging**